src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -30,6 +31,28 @@ public partial class CSharpCompilation
         /// </summary>
         private Symbol[] _lazyWellKnownTypeMembers;
 
+        private bool _needsGeneratedIsReadOnlyAttribute_IsFrozen;
+        private bool _needsGeneratedIsReadOnlyAttribute_Value;
```

Comment:
Unfortunately yes. We need the one in `Compilation` because that's the only thing available when we're not emitting (just producing diagnostics or using the semantic model), and the other one is needed for things that are not declarations on the compilation, like local functions and lambdas. These cannot modify the compilation as they might not belong to it.

---
In reply to: [115036901](https://github.com/dotnet/roslyn/pull/18715#discussion_r115036901) [](ancestors = 115036901)